packages/tracing-internal/src/node/integrations/express.ts

```diff
@@ -317,8 +319,33 @@ function instrumentRouter(appOrRouter: ExpressRouter): void {
       req._hasParameters = true;
     }
 
+    /**
+     * prevent duplicate segment in _reconstructedRoute param if router match multiple routes before final path
+     * example:
+     * original url: /api/v1/1234
+     * prevent: /api/api/v1/:userId
+     * router structure
+     * /api -> middleware
+     * /api/v1 -> middleware
+     * /1234 -> endpoint with param :userId
+     * final _reconstructedRoute is /api/v1/:userId
+     */
+    const originalUrlSplit = req.originalUrl?.split('/').filter(v => !!v);
+    let tempCounter = 0;
+    const currentOffset = req._reconstructedRoute.split('/').filter(v => !!v).length || 0;
+    const layerPath = layer.path
+      ?.split('/')
+      .filter(segment => {
+        if (originalUrlSplit?.[currentOffset + tempCounter] === segment) {
+          tempCounter += 1;
+          return true;
+        }
+        return false;
+      })
+      .join('/');
```

Comment:
Can we extract this into a function and unit test just this logic? I'd like to see a couple different `layer.path` inputs tested to make sure we using offset correctly. This way it's also lazy (we only compute `getNormalizedLayerPath` if the boolean conditional of `layerRoutePath` fails.

```
const partialRoute = layerRoutePath || getNormalizedLayerPath(layer.path, req) || '';
```